gammapy/background/models.py

```diff
@@ -71,3 +77,543 @@ def evaluate(self, x, y):
+
+        for ii in range(n_det_bins_x):
+            if do_only_1_plot:
+                ii = det_bin[0]
+            print("ii", ii)
+            for jj in range(n_det_bins_y):
+                if do_only_1_plot:
+                    jj = det_bin[1]
+                print(" jj", jj)
+                data = self.background[:, ii, jj]
+                detx_bin_center = det_bin_centers[0, ii]
+                dety_bin_center = det_bin_centers[0, jj]
+                print ( "  image({}) canvas({}) pad({})".format(count_images, count_canvases, count_pads))
+
+                if do_only_1_plot:
+                    fig.set_size_inches(8., 8., forward=True)
+                    ax = axes
+                else:
+                    ax = axes.flat[count_pads - 1]
+                image = ax.plot(energy_points.to('TeV'), data, drawstyle='default') # connect points with lines
+                ax.loglog() # double log scale # slow!
+                if do_only_1_plot:
+                    ss_detx_bin_edges = "[{0:.1f}, {1:.1f}) {2}".format(det_bin_edges[0].value, det_bin_edges[1].value, det_bin_edges.unit)
+                    ss_dety_bin_edges = "[{0:.1f}, {1:.1f}) {2}".format(det_bin_edges[2].value, det_bin_edges[3].value, det_bin_edges.unit)
+
+                    ax.set_title('Det = {0} {1}'.format(ss_detx_bin_edges, ss_dety_bin_edges))
+                else:
+                    ss_det_bin_center = "({0:.1f}, {1:.1f})".format(detx_bin_center, dety_bin_center)
+                    ax.set_title('Det = {}'.format(ss_det_bin_center))
+                ax.set_xlabel('E / {}'.format(energy_points.unit))
+                ax.set_ylabel('Bg rate / {}'.format(data.unit))
+                count_pads += 1 # increase
+
+                if count_pads > npads_per_canvas or count_images == nimages:
+                    print("Canvas full, saving and creating a new canvas")
+                    if do_only_1_plot:
+                        filename = "cube_background_model_spectrum{0}{2}{1}{2}.png".format(det.value[0], det.value[1], det.unit)
+                    else:
+                        filename = "cube_background_model_spectra{}.png".format(count_canvases)
+                    print('Writing {}'.format(filename))
+                    fig.savefig(filename)
+                    if not do_only_1_plot:
+                        plt.close('all') # close all open figures
+                        fig, axes = plt.subplots(nrows=nrows, ncols=ncols)
+                        fig.set_size_inches(25., 25., forward=True)
+                    count_canvases += 1 # increase
+                    count_pads = 1 # reset
+
+                count_images += 1 # increase
+
+        if do_only_1_plot:
+            return fig, axes, image
+
+
+    def write(self, filename):
+        """Write cube background model to fits file.
+
+        Parameters
+        ----------
+        filename : `~string`
+            name of file for the bg cube
+        """
+        # number of fields
```

Comment:
Ok, done. Thanks for the example: it saved me a lot of time, since there is no documentation about transforming `~astropy.table.Table` objects into fits `BinTableHDU` or `TableHDU` objects in the astropy webpages.